src/reducers/BreachReducer.js

```diff
@@ -20,37 +19,22 @@ export const BreachReducer = (state = initialState(), action) => {
   const { type } = action;
 
   switch (type) {
-    case BREACH_ACTIONS.OPEN_MODAL: {
-      return { ...state, isModalOpen: true };
-    }
-
     case BREACH_ACTIONS.CLOSE_MODAL: {
       return initialState();
     }
 
-    case BREACH_ACTIONS.SET_BATCH: {
+    case BREACH_ACTIONS.VIEW_ITEM_BREACHES: {
       const { payload } = action;
-      const { batch } = payload;
+      const { breaches, itemName } = payload;
 
-      return { ...state, batch, forBatch: true };
+      return { ...state, breaches, itemName, forItem: true, isModalOpen: true };
     }
 
-    case BREACH_ACTIONS.SET_BATCH_BREACH: {
+    case BREACH_ACTIONS.VIEW_FRIDGE_BREACHES: {
       const { payload } = action;
-      const { batch } = payload;
-
-      return { ...state, batch, forBatch: true };
-    }
-
-    case BREACH_ACTIONS.SET_FRIDGE_BREACH: {
-      const { payload } = action;
-      const { breachId } = payload;
-
-      const breach = UIDatabase.get('TemperatureBreach', breachId) ?? {};
-
-      const { location: fridge } = breach;
+      const { breaches, fridgeName } = payload;
```

Comment:
just simplified this reducer so there are three actions: `viewTransactionItemBreaches`, `viewStocktakeBatchBreaches` and `viewFridgeBreaches`, and just the two reducer cases: `viewItemBreaches` and `viewFridgeBreaches` (mainly for setting the title to be "Breaches for fridge xxx" and `"Breaches for Item xx"`....